app.rb

```diff
@@ -0,0 +1,167 @@
+require_relative 'book'
+require_relative 'teacher'
+require_relative 'student'
+require_relative 'rental'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def run
+    puts 'Welcome to School Library App!'
+    menu
+  end
+
+  def menu
+    loop do
+      puts ' '
+      puts 'Please choose an option by entering a number:'
+      puts '1 - List all books'
+      puts '2 - List all people'
+      puts '3 - Create a person'
+      puts '4 - Create a book'
+      puts '5 - Create a rental'
+      puts '6 - List all rentals for a given person id'
+      puts '7 - Exit'
+      print ''
+
+      choice = gets.chomp.to_i
+
+      case choice
+      when 1..6 then options(choice)
+      when 7 then puts 'Thank you for using our app.'
+      end
+      break if choice == 7
+    end
+  end
+
+  def options(choice)
+    case choice
+    when 1
+      list_books
+    when 2
+      list_people
+    when 3
+      create_person
+    when 4
+      create_book
+    when 5
+      create_rental
+    when 6
+      list_rentals
+    end
+  end
+
+  def list_books
+    if @books.empty?
+      puts 'Sorry, No books available.'
+    else
+      @books.each do |book, index|
+        puts "#{index} - Title: #{book.title}, Author: #{book.author}"
+      end
+    end
+  end
+
+  def list_people
+    if @people.empty?
+      puts 'Sorry, No people available.'
+    else
+      @people.each_with_index do |person, index|
+        puts "#{index}- [#{person.type}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      end
+    end
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    type = gets.chomp.to_i
+    case type
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      puts 'Invalid option. Please enter a valid number: 1 or 2'
+    end
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    permission = gets.chomp
+    person = Student.new(age, name, permission)
+    @people.push(person)
+    puts "Student '#{name}' created successfully"
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Specialization: '
+    specialization = gets.chomp
+    person = Teacher.new(specialization, name, age)
+    @people.push(person)
+    puts "Teacher '#{name}' created successfully"
+  end
```

Comment:
![image](https://github.com/Anand8317/library/assets/91486791/76e4027e-0fb9-4c37-86e9-f747f4b8c95a)
- [ ] All of the people shares the same ID, this would result in error when you search rental by ID. Please change this. Also, your teacher name shows the age, and age show the specialization. Please fix this order as well.

Reply:
Hi, Have updated my code, every person has a unique id now and have fixed the order as well. 
Thank you for pointing out.